### PR TITLE
test: refine func signature of distributed execution test context

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Problem Summary:

Since `NewDistExecutionTestContext` takes the argument `testing.TB`, we can assert the error is not null instead of returning it to the caller.

### What is changed and how it works?

- Refine the function signature of `DistExecutionTestContext`.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by ywqzzy** [comment]
lgtm

File: testkit/mockstore.go [hunk 1]
```diff
@@ -76,26 +75,29 @@ type DistExecutionTestContext struct {
...
 }
 
 // SetOwner set one mock domain to DDL Owner by idx.
-func (d *DistExecutionTestContext) SetOwner(idx int) error {
+func (d *DistExecutionTestContext) SetOwner(idx int) {
 	d.mu.Lock()
 	defer d.mu.Unlock()
 	if idx >= len(d.domains) || idx < 0 {
-		return errors.New("server idx out of bound")
+		d.t.Fatal("server idx out of bound")
```
Comment (zimulala) [inline testkit/mockstore.go:93]: Could we can use `require` too?

File: testkit/mockstore.go [hunk 2]
```diff
@@ -109,25 +111,25 @@ func (d *DistExecutionTestContext) AddServer() {
 }
 
 // DeleteServer delete 1 server by idx, set server0 as ddl owner if the deleted owner is ddl owner.
-func (d *DistExecutionTestContext) DeleteServer(idx int) error {
+func (d *DistExecutionTestContext) DeleteServer(idx int) {
 	d.mu.Lock()
 	defer d.mu.Unlock()
 	if idx >= len(d.domains) || idx < 0 {
-		return errors.New("server idx out of bound")
+		d.t.Fatal("server idx out of bound")
```
Comment (zimulala) [inline testkit/mockstore.go:118]: Could we use `require` too?

File: testkit/mockstore.go [hunk 3]
```diff
@@ -109,25 +111,25 @@ func (d *DistExecutionTestContext) AddServer() {
...
+		return
 	}
 	if len(d.domains) == 1 {
-		return errors.New("can't delete server, since server num = 1")
+		d.t.Fatal("can't delete server, since server num = 1")
```
Comment (zimulala) [inline testkit/mockstore.go:122]: Ditto.

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**